components/suggestion/modal_suggestion_list.tsx

```diff
@@ -1,22 +1,58 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import SuggestionList from 'components/suggestion/suggestion_list.jsx';
 import {getClosestParent} from 'utils/utils';
 
-export default class ModalSuggestionList extends React.PureComponent {
-    static propTypes = {
-        position: PropTypes.string.isRequired,
-        open: PropTypes.bool.isRequired,
-        cleared: PropTypes.bool.isRequired,
-        inputRef: PropTypes.object.isRequired,
-        onLoseVisibility: PropTypes.func.isRequired,
-    }
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface SuggestionItem {}
+
+type Props = {
+    open: boolean;
+    cleared: boolean;
+    inputRef: React.RefObject<HTMLInputElement>;
+    onLoseVisibility: () => void;
+    ariaLiveRef?: React.Ref<HTMLDivElement>;
+    position?: 'top' | 'bottom';
+    renderDividers?: string[];
+    renderNoResults?: boolean;
+    onCompleteWord: (term: string, matchedPretext: string, e?: React.KeyboardEventHandler<HTMLDivElement>) => boolean;
+    preventClose?: () => void;
+    onItemHover: (term: string) => void;
+    pretext: string;
+    matchedPretext: string[];
+    items: SuggestionItem[];
+    terms: string[];
+    selection: string;
+    components: Array<React.Component<{item: SuggestionItem}>>;
+    wrapperHeight?: number;
+
+    // suggestionBoxAlgn is an optional object that can be passed to align the SuggestionList with the keyboard caret
+    // as the user is typing.
+    suggestionBoxAlgn?: {
+        lineHeight: number;
+        pixelsToMoveX: number;
+        pixelsToMoveY: number;
+    };
```

Comment:
where these new props came from. i dont see them being used in the component

Reply:
most of them do not seems to be used in the component

Reply:
@M-ZubairAhmed, **SuggestionList.tsx** component is returned from **modal_suggestion_list.tsx**.
**SuggestionList.tsx** requires these props, that's why their types are defined in here.
Either we make'em optional in **SuggestionList.tsx** or this.

Reply:
(suggestion) If that's the case, and since `suggestion_list` hasn't been migrated yet, would having those as a separate interface (eg `SuggestionListProps`) in here and having Props be something like SuggestionListProps & {...} to make the need for those props explicit (plus they can be moved to `suggestion_list` during its migration?

Reply:
The file containing `SuggestionList` itself hasn't been migrated, but it should have type definitions at least since I added `suggestion_list.d.ts` a while back. You should be able to simplify this by reusing the type definitions then by writing this like
```ts
type Props = React.ComponentProps<SuggestionList> & {
    // any props specific to ModalSuggestionList
}
```

Reply:
We might be able to consolidate these better down the line, but I think we can wait until the rest of this code is migrated to do that. Or we can just leave it until we fix the fact that this uses inheritance since that makes typing this code hard (my bad)